libkbfs/disk_block_cache_test.go

```diff
@@ -687,3 +709,37 @@ func TestDiskBlockCacheUnsyncTlf(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, numBlocks-numBlocksPerTlf, standardCache.numBlocks)
 }
+
+func TestDiskBlockCacheMoveBlock(t *testing.T) {
+	t.Parallel()
+	t.Log("Test that blocks can be moved between caches.")
+	cache, config := initDiskBlockCacheTest(t)
+	defer shutdownDiskBlockCacheTest(cache)
+
+	ctx := context.Background()
+	tlf1 := tlf.FakeID(0, tlf.Private)
+	block1Ptr, _, block1Encoded, block1ServerHalf := setupBlockForDiskCache(
+		t, config)
+
+	t.Log("Put a block into the default cache.")
+	err := cache.Put(
+		ctx, tlf1, block1Ptr.ID, block1Encoded, block1ServerHalf,
+		DiskBlockAnyCache)
+	require.NoError(t, err)
+	err = cache.UpdateMetadata(ctx, block1Ptr.ID, TriggeredPrefetch)
+	require.NoError(t, err)
+	require.Equal(t, 1, cache.workingSetCache.numBlocks)
+	require.Equal(t, 0, cache.syncCache.numBlocks)
+
+	t.Log("Move the block by getting it with a different preferred cache.")
+	_, _, _, err = cache.Get(ctx, tlf1, block1Ptr.ID, DiskBlockSyncCache)
+	require.NoError(t, err)
+	require.Equal(t, 1, cache.syncCache.numBlocks)
+
```

Comment:
Do we want a `require.Equal(t, 0, cache.workingSetCache.numBlocks)` here?

Reply:
Ideally yes, but the Delete happens in another goroutine and I didn't think it was worth adding a bunch of code to track outstanding deletes.